tubular/mapping.py

```diff
@@ -181,9 +196,49 @@ def transform(self, X):
             Transformed input X with levels mapped accoriding to mappings dict.
 
         """
+        mapped_columns = self.mappings.keys()
+        original_dtypes = X[mapped_columns].dtypes
+
+        for col in mapped_columns:
+
+            values_to_be_mapped = set(self.mappings[col].keys())
+            values_in_df = set(X[col].unique())
+
+            if len(values_to_be_mapped.intersection(values_in_df)) == 0:
+
+                warnings.warn(
+                    f"{self.classname()}: No values from mapping for {col} exist in dataframe."
+                )
+
+            if len(values_to_be_mapped.difference(values_in_df)) > 0:
+
+                warnings.warn(
+                    f"{self.classname()}: There are values in the mapping for {col} that are not present in the dataframe"
+                )
 
         X = BaseMappingTransformMixin.transform(self, X)
 
+        mapped_dtypes = X[mapped_columns].dtypes
+
+        for col in mapped_columns:
```

Comment:
I would add a `suppress_dtype_warning` parameter (default `False`) to this transformer - if you're applying a mapping from e.g. integer to bool, you know that the dtype is going to change and this warning would be quite annoying

Reply:
good call, have added this in [e9c8d66](https://github.com/lvgig/tubular/pull/54/commits/e9c8d6687626a87102295d53ffb1f34ccd916362)